packages/editor/src/components/document-bar/index.js

```diff
@@ -48,39 +50,75 @@ const icons = {
 };
 
 export default function DocumentBar() {
-	const { isEditingTemplate, templateId, postType, postId } = useSelect(
+	const patternId = parseInt(
+		getQueryArg( window.location.href, 'patternId' )
+	);
+
+	const {
+		isEditingTemplate,
+		templateId,
+		postType,
+		postId,
+		isEditingPattern,
+		dirtyEntityRecords,
+	} = useSelect(
 		( select ) => {
 			const {
 				getRenderingMode,
 				getCurrentTemplateId,
 				getCurrentPostId,
 				getCurrentPostType,
 			} = select( editorStore );
+			const { __experimentalGetDirtyEntityRecords } =
+				select( coreDataStore );
 			const _templateId = getCurrentTemplateId();
+			const renderingMode = getRenderingMode();
 			return {
 				isEditingTemplate:
-					!! _templateId && getRenderingMode() === 'template-only',
+					!! _templateId && renderingMode === 'template-only',
+				isEditingPattern: renderingMode === 'pattern-only',
 				templateId: _templateId,
-				postType: getCurrentPostType(),
-				postId: getCurrentPostId(),
+				postType:
+					renderingMode !== 'pattern-only'
+						? getCurrentPostType()
+						: 'wp_block',
+				postId:
+					renderingMode !== 'pattern-only'
+						? getCurrentPostId()
+						: patternId,
+				dirtyEntityRecords: __experimentalGetDirtyEntityRecords(),
 			};
 		},
-		[]
+		[ patternId ]
 	);
 	const { getEditorSettings } = useSelect( editorStore );
 	const { setRenderingMode } = useDispatch( editorStore );
+	const { createWarningNotice } = useDispatch( noticesStore );
+
+	function handleOnBack() {
+		if (
+			patternId &&
+			dirtyEntityRecords.length > 0 &&
+			dirtyEntityRecords.find( ( record ) => record.key === patternId )
+		) {
+			createWarningNotice(
+				__(
+					'Your pattern changes will not show in the post until they are saved'
```

Comment:
We should be able to fix this. I imagine it's because since #56235, the experimental version of the pattern block no longer uses `useEntityBlockEditor`, the inner blocks are static when the block is first created and only use the pattern data as a sort of 'starter content'.

That's an unanticipated problem (cc @kevin940726).

Reply:
Nice catch! I opened a PR here: https://github.com/WordPress/gutenberg/pull/57148.